From 54f1e01311592927ddd23ec18ae5276b0e355f3e Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 31 Mar 2022 18:53:50 +0000
Subject: [PATCH] libstdc++: fix incomplete type of ModelProvider

Destructor of std::unique_ptr in libstdc++ uses sizeof() which
requires full definition of segmentation_platform::ModelProvider
in segmentation_platform::GetSegmentationDefaultModelProvider().

Bug: 957519
Change-Id: Icf70536550f962702be827ad2676e5d3e5d9d489
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3563162
Reviewed-by: Shakti Sahu <shaktisahu@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#987566}
---
 .../segmentation_platform/segmentation_platform_config.cc        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/chrome/browser/segmentation_platform/segmentation_platform_config.cc b/chrome/browser/segmentation_platform/segmentation_platform_config.cc
index b2bf48ed30aa2..1c817aeb91761 100644
--- a/chrome/browser/segmentation_platform/segmentation_platform_config.cc
+++ b/chrome/browser/segmentation_platform/segmentation_platform_config.cc
@@ -12,6 +12,7 @@
 #include "components/optimization_guide/proto/models.pb.h"
 #include "components/segmentation_platform/public/config.h"
 #include "components/segmentation_platform/public/features.h"
+#include "components/segmentation_platform/public/model_provider.h"
 
 #if BUILDFLAG(IS_ANDROID)
 #include "base/metrics/field_trial_params.h"
-- 
2.35.1

